### PR TITLE
docs: add more content to concepts chapter

### PR DESCRIPTION
**Pull request by nicecui**

### CODE REVIEW

File: docs/user-guide/concepts.md [hunk 1]
```diff
@@ -1,8 +1,61 @@
 # Concepts
 
-![architecture](../public/architecture-2.png)
+<!-- ## Why GreptimeDB 
+Articl summary and link to blog post -->
```
Comment (xtang) [inline docs/user-guide/concepts.md:4]:
```suggestion
Article summary and link to blog post -->
```

File: docs/user-guide/concepts/why-greptimedb.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Why GreptimeDB
```
Comment (nicecui) [inline docs/user-guide/concepts/why-greptimedb.md:1]: This document has reviewed by @apdong2022

File: docs/user-guide/concepts/storage-location.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Storage Location
```
Comment (nicecui) [inline docs/user-guide/concepts/storage-location.md:1]: @killme2008 PTAL

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by xtang — APPROVED** [review]

**Pull request merged**